src/vehicle_base.h

```diff
@@ -327,6 +337,8 @@ struct Vehicle : VehiclePool::PoolItem<&_vehicle_pool>, BaseVehicle, BaseConsist
 	NewGRFCache grf_cache;              ///< Cache of often used calculated NewGRF values
 	VehicleCache vcache;                ///< Cache of often used vehicle values.
 
+	MutableSpriteCache sprite_cache;    ///< Cache of sprites and values related to recalculating them, see #MutableSpriteCache
```

Comment:
I might have put the `mutable` here instead of several inside the struct, but that is a bit bikeshedding.

Reply:
I did ponder the same, it's an easy change so I'll make it. Otherwise someone might find it unexpected that a "Mutable" sprite cache has a non-mutable member.